src/vm_pmu.rs

```diff
@@ -373,6 +373,112 @@ impl VmPmuState {
         }
     }
 
+    // Implementation of PMU counter configuration on platforms with Sscdeleg support
+    // TODO: Use code in the actual implementation
+    #[allow(dead_code)]
+    fn configure_matching_counters_internal(
+        &mut self,
+        counter_index: u64,
+        counter_mask: u64,
+        config_flags: PmuCounterConfigFlags,
+        _event_type: PmuEventType,
+        _event_data: u64,
+    ) -> SbiResult<u64> {
+        // TODO: Finalize the logic for using event_type and event_data
+        use PmuCounterState::*;
+        let pmu_info = pmu::PmuInfo::get()?;
+        let counter_mask = pmu_info.filter_counter_mask(counter_index, counter_mask)?;
+        const FIXED_COUNTERS_MASK: u64 = !0x7;
+        let index = if !config_flags.is_skip_match() {
+            let mut bitmask_iter =
+                CounterMaskIter::new(counter_index, counter_mask & FIXED_COUNTERS_MASK);
```

Comment:
`counter_mask` is relative to `counter_index`, so if the intention is to mask out the fixed counters this logic is incorrect if `counter_index` != 0.

But also, we do need to support configuring `cycle` and `instret`, right?

Reply:
Right, was unsure about how we wanted to support the fixed counters. Per the conversation with Atish, it's best to just use fixed counters for 0, 2 (instead of attempting the SBI approach of trying to mapping them to programmable counters).